Models/ProductsListPart.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Orchard.ContentManagement;
+using Orchard.ContentManagement.Utilities;
+using Orchard.Environment.Extensions;
+using Orchard.Security;
+using Newtonsoft.Json.Linq;
+
+namespace Nwazet.Commerce.Models {
+    [OrchardFeature("Nwazet.PersistentCart")]
+    public class ProductsListPart : ContentPart {
+        //This part goes in the infoset
+
+        private List<ShoppingCartItem> _items {
+            get {
+                var fromInfoset = Retrieve<string>("SerializedItems");
```

Comment:
It's wasteful to do this double serialization: serialize an array of objects into a string, that then gets serialized as part of the Infoset? You should be able to deal directly with the array without doing the serialization and deserialization yourself.